### PR TITLE
FOUNDATIONS: Add Refugee

### LINKED ISSUE #9: FOUNDATIONS: Add Refugee

### PR DESCRIPTION
**Pull request by ElbekDeveloper**
Closes #9

### CODE REVIEW

File: RefugeeLand.Core.Api.Tests.Unit/Foundations/RefugeeServiceTests.Exceptions.Add.cs [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+                        expectedRefugeeDependencyValidationException))),
+                Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                    broker.InsertRefugeeAsync(It.IsAny<Refugee>()),
+                Times.Never);
+
+            this.dateTimeBrokerMock.Verify(broker =>
+                    broker.GetCurrentDateTimeOffset(),
+                Times.Once);
```
Comment (ElbekDeveloper) [inline RefugeeLand.Core.Api.Tests.Unit/Foundations/RefugeeServiceTests.Exceptions.Add.cs:97]: The business logic calls `dateTimeBroker` to throw exception, then `loggingBroker` is called. So this should be the order of `.Verify()` statements. The same applies to `VerifyNoOtherCalls()`

File: RefugeeLand.Core.Api.Tests.Unit/Foundations/RefugeeServiceTests.Exceptions.Add.cs [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
```
Comment (ElbekDeveloper) [inline RefugeeLand.Core.Api.Tests.Unit/Foundations/RefugeeServiceTests.Exceptions.Add.cs:101]:
```cs
this.dateTimeBrokerMock.VerifyNoOtherCalls();
this.loggingBrokerMock.VerifyNoOtherCalls();
this.storageBrokerMock.VerifyNoOtherCalls();
```

File: RefugeeLand.Core.Api.Tests.Unit/Foundations/RefugeeServiceTests.Exceptions.Add.cs [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+                    broker.InsertRefugeeAsync(It.IsAny<Refugee>()),
+                        Times.Never);
+
+            this.dateTimeBrokerMock.Verify(broker =>
+                    broker.GetCurrentDateTimeOffset(),
+                        Times.Once);
+
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
```
Comment (ElbekDeveloper) [inline RefugeeLand.Core.Api.Tests.Unit/Foundations/RefugeeServiceTests.Exceptions.Add.cs:143]: Please, do the same for all of the tests

**Review by bramburn — COMMENTED** [review]
@ElbekDeveloper why would tests fall under foundations? would you not use something like TEST:?

**Comment by Spectralgo** [comment]
@bramburn Because those tests are part of TDD, the only way to build a foundation service is through those tests.

**Pull request merged**